rstudio/rstudio
# Bugfix/electron logging support

### Intent
Address #10929 and #11565

### Approach
Reads from `logging.conf` and environment variables. There's an object now to represent the options and maps the command-line/logging.conf option to the environment variable.

Now that `logger-type` can be set to `stderr`, the log to console has been removed from dev builds.

Syslog has been implemented for Linux only. OSX has a syslog but it doesn't appear well documented and `winston-syslog` doesn't seem able to log to it. It might be `local0` isn't supported and `udp` must be used but changing it doesn't appear to be allowed without disabling some security permissions.

### Automated Tests
There are some unit tests to ensure the log options are set properly. It doesn't check that it can log with those options though.

### QA Notes
Linux should log to `syslog` without any configuration. It's set to use local sockets so there shouldn't be any requirement to configure a firewall.

`stderr` will be visible when running RStudio from a command-line.

`file` is the default logger type if a type is unspecified. It still follows a daily rotated log file with the name including the date.

### Checklist
